doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -185,129 +166,118 @@ Newly opened modules and sections inherit the current settings.
-      notation’s string denoting some reference :n:`@qualid` (referred to by its
-      main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
-      `"_ 'U' _"`, see Section :ref:`notations`), the command works like ``Search`` :n:`@qualid`.
+.. todo: move Search* to the section on query commands once we agree on the wording
 
-   .. cmdv:: Search @string%@ident
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      The string string must be a notation or the main
-      symbol of a notation which is then interpreted in the scope bound to
-      the delimiting key :token:`ident` (see Section :ref:`LocalInterpretationRulesForNotations`).
+   .. insertprodn search_item search_item
 
-   .. cmdv:: Search @term_pattern
+   .. prodn::
+      search_item ::= {? - } @string {? @scope }
+      | {? - } @one_term
 
-      This searches for all statements or types of
-      definition that contains a subterm that matches the pattern
-      :token:`term_pattern` (holes of the pattern are either denoted by `_` or by
-      :n:`?@ident` when non linear patterns are expected).
+   Todo: :n:`@scope ::= % @ident` (fix in syntax extensions)
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string}
+   Displays the name and type of objects matching :n:`@search_item`\s
+   in the selected goals (if a proof is open)
+   as well as accessible theorems, axioms, etc..
+   It's useful for finding the names of library lemmas.
 
-      where
-      :n:`@term_pattern_string` is a term_pattern, a string, or a string followed
-      by a scope delimiting key `%key`.  This generalization of ``Search`` searches
-      for all objects whose statement or type contains a subterm matching
-      :n:`@term_pattern` (or :n:`@qualid` if :n:`@string` is the notation for a reference
-      qualid) and whose name contains all string of the request that
-      correspond to valid identifiers. If a term_pattern or a string is
-      prefixed by `-`, the search excludes the objects that mention that
-      term_pattern or that string.
+   * :n:`@string` - If :n:`@string` is a substring of a valid identifier,
+     search for objects whose name contains :n:`@string`. If :n:`@string` is a notation
+     string associated with a :n:`@qualid`, that's equivalent to :cmd:`Search` :n:`@qualid`.
+     For example, specifying "+" or "_ + _", which are notations for "plus", are equivalent
+     to :cmd:`Search` :n:`plus`.
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string} inside {+ @qualid }
+   * :n:`@scope` - limits the search to the scope bound to
+     the delimiting key :n:`@scope`, such as, for example, :n:`%nat`
+     (see Section :ref:`LocalInterpretationRulesForNotations`).
 
-      This restricts the search to constructions defined in the modules
-      named by the given :n:`qualid` sequence.
+   If you specify multiple :n:`@search_item`\s, all the conditions must be satisfied
+   for the object to be displayed.
```

Comment:
There's no reason that this appears before the explanation of the `one_term` kind of search_item.